packages/dds/tree/src/test/utils.ts

```diff
@@ -26,3 +39,110 @@ export class MockDependent extends SimpleObservingDependent {
 		super((token) => this.tokens.push(token), name);
 	}
 }
+
+/**
+ * A test helper class that manages the creation, connection and retrieval of SharedTrees. Instances of this
+ * class are created via {@link create} and satisfy the {@link ITestObjectProvider} interface.
+ */
+export class TestTreeProvider {
+    private static readonly treeId = "TestSharedTree";
+
+    [tree: number]: SharedTree;
+
+    private readonly provider: ITestObjectProvider;
+    private readonly trees: SharedTree[] = [];
+
+    /**
+     * Create a new {@link TestTreeProvider}. The provider can be populated with trees via subsequent calls to
+     * {@link createTree}. If the number of trees used by a test is known ahead of time, consider using
+     * {@link createTrees} instead.
+     */
+    public static async create(): Promise<TestTreeProvider & ITestObjectProvider> {
+        return new TestTreeProvider() as TestTreeProvider & ITestObjectProvider;
+    }
+
+    /**
+     * Create a new {@link TestTreeProvider} with a number of trees pre-initialized.
+     * @param trees - the number of trees to initialize this provider with. This is the same as calling
+     * {@link create} followed by {@link createTree} _trees_ times.
+     *
+     * @example
+     * ```ts
+     * const trees = await TestTreeProvider.createTrees(2);
+     * assert(trees[0].isAttached());
+     * assert(trees[1].isAttached());
+     * await trees.ensureSynchronized();
+     * ```
+     */
+    public static async createTrees(trees: number): Promise<TestTreeProvider & ITestObjectProvider> {
+        const provider = await this.create();
+        for (let i = 0; i < trees; i++) {
+            await provider.createTree();
+        }
+        return provider;
+    }
```

Comment:
Since 'createTrees(0)' is the same as 'create'.  You could combine these.

Reply:
I was trying to support two different flows here, but without actually making two different classes.

1. All trees are known ahead of time and made at the start of the test, and interaction with the `provider` object itself is probably limited.
```ts
const trees = TestTreeProvider.createTrees(3);
assertTreesAreEqual(trees[0], trees[1]);
trees.ensureSynchronized(); // This is a method on the provider, but the way the words flow, it works.
assertTreesAreEqual(trees[1], trees[2]);
...
```

2. Trees come into being over time, or the `provider` object is interacted with directly a lot.
```ts
const provider = TestTreeProvider.create();
const treeBeforeSync = await provider.createTree();
await provider.ensureSynchronized();
const treeAfterSync = await provider.createTree();
provider.loadContainer(); // This is a method on the provider, it would be weird to do this to an object called "trees".
...
```

We had a mix of these kinds of tests in legacy shared tree, so this was my attempt at being flexible and still keeping syntax pretty short. What do you think? Does it help, or is it overcomplicated?

Reply:
I was trying to support two different flows here, but without actually making two different classes.

1. All trees are known ahead of time and made at the start of the test, and interaction with the `provider` object itself is probably limited.
```ts
const trees = TestTreeProvider.createTrees(3);
assertTreesAreEqual(trees[0], trees[1]);
trees.ensureSynchronized(); // This is a method on the provider, but the way the words flow, it works.
assertTreesAreEqual(trees[1], trees[2]);
...
```

2. Trees come into being over time, or the `provider` object is interacted with directly a lot.
```ts
const provider = TestTreeProvider.create();
const treeBeforeSync = await provider.createTree();
await provider.ensureSynchronized();
const treeAfterSync = await provider.createTree();
provider.loadContainer(); // This is a method on the provider, it would be weird to do this to an object called "trees".
...
```

We had a mix of these kinds of tests in legacy shared tree, so this was my attempt at being flexible and still keeping syntax pretty short. What do you think? Does it help, or is it overcomplicated?